### PR TITLE
wordcheck_engine: Use DPDatabase APIs and use PHP7 array syntax

### PR DESCRIPTION
**Pull request by bpfoley**
Sandbox in https://www.pgdp.org/~bfoley/c.branch/cleanup18/

Tested by going through various permutations of the wordcheck interface.

### CODE REVIEW

File: pinc/wordcheck_engine.inc [hunk 1]
```diff
@@ -378,23 +378,23 @@ function get_bad_words_via_pattern($input_words_w_freq, $languages)
...
-    );
-    $exceptions = array();
+    ];
+    $exceptions = [];
     foreach($langcode3s as $langcode3)
     {
         $exceptions[] = array_get($exceptions_for_lang, $langcode3, "");
     }
-    $exceptions = implode('|', $exceptions);
+    $exceptions = join('|', $exceptions);
```
Comment (chrismiceli) [inline pinc/wordcheck_engine.inc:397]: why this change?

File: pinc/wordcheck_engine.inc [hunk 2]
```diff
@@ -1103,15 +1100,7 @@ function get_langcode3s_for_languages($languages)
...
-    $natcaseresult = strnatcasecmp($a, $b);
-    if($natcaseresult != 0)
-    {
-        return $natcaseresult;
-    }
-    else
-    {
-        return strnatcmp($a, $b);
-    }
+    return strnatcasecmp($a, $b)?:strnatcmp($a, $b);
```
Comment (chrismiceli) [inline pinc/wordcheck_engine.inc:1103]: I like the explicit check to 0 that we had before instead of relying on it being falsy.

File: pinc/wordcheck_engine.inc [hunk 3]
```diff
@@ -563,44 +565,36 @@ function copy_project_wordcheck_events($from_projectid, $to_projectid)
...
 function save_wordcheck_event($projectid, $round, $page, $proofer, $suggestions, $corrections)
 {
-    $suggestion_list = implode("\n", $suggestions);
-
-    $correction_list = array();
-
     // save the corrections in wdiff format
-    foreach($corrections as $correction) {
-        list($orig, $new) = $correction;
+    $correction_list = [];
```
Comment (chrismiceli) [inline pinc/wordcheck_engine.inc:581]: is this used anymore?

Comment (bpfoley) [reply pinc/wordcheck_engine.inc:581]: Well spotted. It looks like not. We're inserting and reading it from the DB, but as far as I can see the only user of `load_wordcheck_events` in `show_good_word_suggestions_detail.php` doesn't use it anymore. I'd prefer to do this as a followup MR so as not to change any behavior in this one though.

File: pinc/wordcheck_engine.inc (hunk 2 above)
Comment (bpfoley) [reply pinc/wordcheck_engine.inc]: OK. Perhaps I'm too used to false-y 0 in C and Python.

File: pinc/wordcheck_engine.inc (hunk 1 above)
Comment (bpfoley) [reply pinc/wordcheck_engine.inc:397]: No great reason. I just trip over `implode` every time I see it. Every other language I know that has a similar feature calls it `join` and the word feels less jargony and more pseudocode to me. I figured that since I was doing the cosmetic `array(...)` to `[...]` conversion I may as well do this too. (And of course, PHP being PHP, `split` had to be a different, now deprecated function that isn't the inverse of `join`, so we're stuck with `explode`. Sigh.)

File: pinc/wordcheck_engine.inc [hunk 4]
```diff
@@ -626,12 +620,16 @@ function count_wordcheck_suggestion_events($projectid, $start_time=0)
...
+        SELECT * FROM wordcheck_events
+        WHERE projectid='%s',
+        AND timestamp>%d",
+        DPDatabase::escape($projectid),
+        $start_time
+    );
 
-    $res = mysqli_query(DPDatabase::get_connection(), $sql);
+    $eventArray = [];
+    $res = DPDatabase::query($sql, FALSE);
```
Comment (cpeel) [inline pinc/wordcheck_engine.inc:632]: While this is true to the original, the original is wrong and this should raise on error. If the query returns zero rows (which is possible) `$res` will still be not FALSE.

File: pinc/wordcheck_engine.inc (hunk 1 above)
Comment (cpeel) [reply pinc/wordcheck_engine.inc:397]: I'm neutral on these changes. I'm happy to take them but I don't feel like we need to search/replace to change them.

File: pinc/wordcheck_engine.inc [hunk 5]
```diff
@@ -614,7 +608,7 @@ function count_wordcheck_suggestion_events($projectid, $start_time=0)
             WHERE projectid='$projectid'
```
Comment (cpeel) [inline pinc/wordcheck_engine.inc:608]: We need to escape `$projectid` and ensure `$start_time` is a number via sprintf. Also we should raise an exception on error (the FALSE check is wrong in the original code).

Comment (bpfoley) [reply pinc/wordcheck_engine.inc]: Done.
